cmd/system-probe/loader.go

```diff
@@ -31,8 +33,11 @@ func (l *Loader) Register(cfg *config.AgentConfig, httpMux *http.ServeMux, facto
 			continue
 		}
 
+		// In case a module failed to be started, do not make the whole `system-probe` abort.
+		// Let `system-probe` run the other modules.
 		if err != nil {
-			return errors.Wrapf(err, "new module `%s` error", factory.Name)
+			log.Errorf("new module `%s` error: %v", factory.Name, err)
+			continue
 		}
 
 		if err = module.Register(httpMux); err != nil {
```

Comment:
We do abort the load in this case?

Reply:
As far as module load is concerned, (the lines above) it’s the configuration of the host that can explain that some modules will always fail to start whereas some others will always succeed to start.
Indeed, depending on the kernel version and on whether or not the kernel headers are installed, we can have the offset-guessing-based module that will always fail or the runtime-compilation-based modules that will always fail.

In those cases, it would be better to have `system-probe` able to run the modules that could be loaded.


Here, for `module.Register(httpMux)`, it’s a little bit different because, the [`Register` function](https://github.com/DataDog/datadog-agent/blob/03c6806ad79386b08e58be10b85fc085ffc1851b/pkg/security/module/module.go#L51) completely deletes the UNIX socket before recreating it.
As a consequence, if the `Register` function fails once, it might break all the modules, including the ones that were previously loaded.
In this situation, it may be useless to keep `system-probe` up and running.

Reply:
But this is a general loader function to load all the modules right? The particular Register function for the security probe does this, but other Register functions for other modules may not do this. Is it not ok for the system-probe to continue to run without the security module? If that is the case, maybe we should have Register return a special error that other modules can also use to signal to the loader that the system-probe should abort.

Reply:
You’re right.
I looked a little bit too quickly at what this function was doing for all the modules.
It indeed makes sense to let `system-probe` start if only some modules are failing to `Register`.
I’ve just updated the PR accordingly.